src/ai/backend/manager/models/image.py

```diff
@@ -895,95 +997,125 @@ async def _in_domain_scope(
             object_id_to_additional_permission_map=image_id_permission_map
         )
 
-    async def _in_project_scope(
+    async def _get_user_accessible_project_scopes(
         self,
         ctx: ClientContext,
-        scope: ProjectScope,
-    ) -> ImagePermissionContext:
-        from .group import GroupRow
+        scope: UserScope,
+    ) -> list[ProjectScope]:
+        from .group import AssocGroupUserRow
 
-        permissions = await self.calculate_permission(ctx, scope)
-        image_id_permission_map: dict[UUID, frozenset[ImagePermission]] = {}
+        get_assoc_group_ids_stmt = sa.select(AssocGroupUserRow.group_id).where(
+            AssocGroupUserRow.user_id == scope.user_id
+        )
+        group_ids = await self.db_session.scalars(get_assoc_group_ids_stmt)
+
+        return [ProjectScope(project_id=group_id) for group_id in group_ids]
+
+    async def _verify_project_scope_and_calculate_permission(
+        self, ctx: ClientContext, scope: ProjectScope
+    ) -> frozenset[ImagePermission]:
+        from .group import GroupRow
 
         group_query_stmt = sa.select(GroupRow).where(GroupRow.id == scope.project_id)
         group_row = cast(Optional[GroupRow], await self.db_session.scalar(group_query_stmt))
         if group_row is None:
-            raise InvalidScope(f"Project not found (project_id:{scope.project_id})")
+            raise InvalidScope(f"Project not found (project_id: {scope.project_id})")
+
+        return await self.calculate_permission(ctx, scope)
+
+    async def _in_project_scopes_by_registry_condition(
+        self,
+        ctx: ClientContext,
+        scopes: list[ProjectScope],
+        registry_condition_factory: Callable[[list[Any]], Any],
```

Comment:
I think it would be great if there is a better way than declaring registry_condition_factory and filter_global_registry redundantly.